### PR TITLE
Update identity governance documentation

### PR DESCRIPTION
**Pull request by madurangasiriwardena**
image

### CODE REVIEW

File: en/docs/develop/identity-governance-rest-api.md [hunk 1]
```diff
@@ -3,16 +3,164 @@ template: templates/swagger.html
 ---
 
 # WSO2 Identity Server - governance connectors admin API
+     
+This is the RESTful API for managing identity governance configurations in WSO2 Identity Server. These configurations
+ are grouped in to main 5 categories. Each category will have sub categories which are called connectors, and
```
Comment (gomathyk) [inline en/docs/develop/identity-governance-rest-api.md]: grouped into five main categories. 

Each category has subcategories

File: en/docs/develop/identity-governance-rest-api.md [hunk 2]
```diff
@@ -3,16 +3,164 @@ template: templates/swagger.html
 ---
 
 # WSO2 Identity Server - governance connectors admin API
+     
+This is the RESTful API for managing identity governance configurations in WSO2 Identity Server. These configurations
+ are grouped in to main 5 categories. Each category will have sub categories which are called connectors, and 
+ properties of connectors as mentioned below. <br>
 
-??? Note "Note: Click to view"
+??? "Password Policies"
```
Comment (gomathyk) [inline en/docs/develop/identity-governance-rest-api.md]: ??? note "Password Policies"

File: en/docs/develop/identity-governance-rest-api.md [hunk 3]
```diff
@@ -3,16 +3,164 @@ template: templates/swagger.html
...
+
+    * Password History
+    
+    | Property display name | Property name | Purpose |
+    | ------- | -------|---|
+    | Enable Password History Feature   | passwordHistory.enable   | Enable to disallow previously used passwords |
+    | Password History validation count   | passwordHistory.count   | Restrict reusing last x number of password during password update |
```
Comment (gomathyk) [inline en/docs/develop/identity-governance-rest-api.md]: x number of passwords when updating the password

File: en/docs/develop/identity-governance-rest-api.md [hunk 4]
```diff
@@ -3,16 +3,164 @@ template: templates/swagger.html
...
+    
+    | Property display name | Property name | Purpose |
+    | ------- | -------|---|
+    | Enable Password Policy Feature   | passwordPolicy.enable   | Enable password pattern policy |
+    | Password Policy Min Length   | passwordPolicy.min.length   | Minumum length of the password |
+    | Password Policy Max Length   | passwordPolicy.max.length   | Maximum length of the password |
+    | Password Policy Pattern   | passwordPolicy.pattern   | Allowed password regex pattern |
+    | Password Policy Error Message   | passwordPolicy.errorMsg   | Error message for invalid password patterns |
+
+??? "Login Policies"
```
Comment (gomathyk) [inline en/docs/develop/identity-governance-rest-api.md]: The format for collapsible is-  ??? note "xxx". Please change in all relevant places

File: en/docs/develop/identity-governance-rest-api.md [hunk 5]
```diff
@@ -3,16 +3,164 @@ template: templates/swagger.html
...
+
+    * Account Locking
+    
+    | Property display name | Property name | Purpose |
+    | ------- | -------|---|
+    | Account Lock Enabled   | account.lock.handler.enable   | Enable account locking for failed logins |
+    | Maximum Failed Login Attempts   | account.lock.handler.On.Failure.Max.Attempts   | Number of failed attempts allows without locking the account |
```
Comment (gomathyk) [inline en/docs/develop/identity-governance-rest-api.md]: attempts allowed before locking the account

File: en/docs/develop/identity-governance-rest-api.md [hunk 6]
```diff
@@ -3,16 +3,164 @@ template: templates/swagger.html
...
+    | Property display name | Property name | Purpose |
+    | ------- | -------|---|
+    | Enable Account Disabling   | account.disable.handler.enable   | Enable account disable Feature |
+    | Internal Notification Management   | account.disable.handler.notification.manageInternally   | Set false if the client application handles notification sending |
+    
+    * Captcha for SSO Login
+    
+    | Property display name | Property name | Purpose |
+    | ------- | -------|---|
+    | Enable captcha always   | sso.login.recaptcha.enable.always   | Enable captcha verification always during SSO login |
```
Comment (gomathyk) [inline en/docs/develop/identity-governance-rest-api.md]: Always enable captcha verification during SSO login

File: en/docs/develop/identity-governance-rest-api.md [hunk 7]
```diff
@@ -3,16 +3,164 @@ template: templates/swagger.html
...
+    | Enable captcha after the Max failed attempts   | sso.login.recaptcha.enable   | Enable captcha verification during SSO login after allowed number of failed attempts |
```
Comment (gomathyk) [inline en/docs/develop/identity-governance-rest-api.md]: allowing a certain number of failed attempts

File: en/docs/develop/identity-governance-rest-api.md [hunk 8]
```diff
@@ -3,16 +3,164 @@ template: templates/swagger.html
...
+    | Max failed attempts   | sso.login.recaptcha.on.max.failed.attempts   | Number of failed attempts allows without showing the captcha |
```
Comment (gomathyk) [inline en/docs/develop/identity-governance-rest-api.md]: allowed

File: en/docs/develop/identity-governance-rest-api.md [hunk 9]
```diff
@@ -3,16 +3,164 @@ template: templates/swagger.html
...
+    | ------- | -------|---|
+    | Enable Notification Based Password Recovery   | Recovery.Notification.Password.Enable   |  |
+    | Enable reCaptcha for Password Recovery   | Recovery.ReCaptcha.Password.Enable   |  |
+    | Enable Security Question Based Password Recovery   | Recovery.Question.Password.Enable   |  |
+    | Number Of Questions Required For Password Recovery   | Recovery.Question.Password.MinAnswers   |  |
+    | Enable reCaptcha for Security Questions Based Password Recovery   | Recovery.Question.Password.ReCaptcha.Enable   | Show captcha for challenge question based password recovery |
+    | Max Failed Attempts for ReCaptcha   | Recovery.Question.Password.ReCaptcha.MaxFailedAttempts   |  |
+    | Enable Username Recovery   | Recovery.Notification.Username.Enable   |  |
+    | Enable reCaptcha for Username Recovery   |  Recovery.ReCaptcha.Username.Enable  |  |
+    | Enable Internal Notification Management   | Recovery.Notification.InternallyManage  | Set false if the client application handles notification sending |
```
Comment (gomathyk) [inline en/docs/develop/identity-governance-rest-api.md]: Set to

File: en/docs/develop/identity-governance-rest-api.md [hunk 10]
```diff
@@ -3,16 +3,164 @@ template: templates/swagger.html
...
+    | Enable forced challenge questions   | Recovery.Question.Password.Forced.Enable   | Force users to provide answers to challenge questions during sign in |
+    | Recovery callback URL regex   | Recovery.CallbackRegex   |  |
+    
+    * User Self Registration
+    
+    | Property display name | Property name | Purpose |
+    | ------- | -------|---|
+    | Enable Self User Registration  | SelfRegistration.Enable  | Enable self user registration |
+    | Enable Account Lock On Creation | SelfRegistration.LockOnCreation  | Lock user account during user registration |
+    | Internal Notification Management  | SelfRegistration.Notification.InternallyManage  | Set false if the client application handles notification sending |
```
Comment (gomathyk) [inline en/docs/develop/identity-governance-rest-api.md]: Set to

File: en/docs/develop/identity-governance-rest-api.md [hunk 11]
```diff
@@ -3,16 +3,164 @@ template: templates/swagger.html
...
+    | Enable reCaptcha  | SelfRegistration.ReCaptcha  | Enable captcha verification during self registration |
+    | User self registration verification link expiry time  | SelfRegistration.VerificationCode.ExpiryTime  | Set the number of minutes the user self registration verification mail would be valid |
```
Comment (gomathyk) [inline en/docs/develop/identity-governance-rest-api.md]: Set the number of minutes for which the user self registration verification mail is valid

File: en/docs/develop/identity-governance-rest-api.md [hunk 12]
```diff
@@ -3,16 +3,164 @@ template: templates/swagger.html
...
+    | User self registration callback URL regex  | SelfRegistration.CallbackRegex  | User self registration callback URL regex |
+    | Manage Self-Sign-Up purposes  | _url_listPurposeSelfSignUp  | Click here to manage Self-Sign-Up purposes |
+    
+    * User Onboarding
+    
+    | Property display name | Property name | Purpose |
+    | ------- | -------|---|
+    | Enable User Email Verification  | EmailVerification.Enable  | Enable to trigger a verification notification during user creation |
+    | Enable Account Lock On Creation  | EmailVerification.LockOnCreation  | Lock user account during user creation |
+    | Internal Notification Management  | EmailVerification.Notification.InternallyManage  | Set false if the client application handles notification sending |
```
Comment (gomathyk) [inline en/docs/develop/identity-governance-rest-api.md]: Set to

File: en/docs/develop/identity-governance-rest-api.md [hunk 13]
```diff
@@ -3,16 +3,164 @@ template: templates/swagger.html
...
+    | Email verification code expiry time  | EmailVerification.ExpiryTime  | Set the number of minutes the email verification mail would be valid.(Negative value for infinite validity) |
```
Comment (gomathyk) [inline en/docs/develop/identity-governance-rest-api.md]: Set the number of minutes for which the email verification mail is valid. Setting it to a negative value results in infinite validity.

File: en/docs/develop/identity-governance-rest-api.md [hunk 14]
```diff
@@ -3,16 +3,164 @@ template: templates/swagger.html
...
+    | Ask password code expiry time  | EmailVerification.AskPassword.ExpiryTime  | Set the number of minutes the ask password mail would be valid. (Negative value for infinite validity) |
+    | Temporary password generation extension class  | EmailVerification.AskPassword.PasswordGenerator  | Temporary password generation extension point in ask password feature) |
+    | Manage Just In Time Provisioning purposes  | _url_listPurposeJITProvisioning  | Click here to manage JIT purposes |
+    
+    * Account Management Policies
+    
+    | Property display name | Property name | Purpose |
+    | ------- | -------|---|
+    | Enable Password Reset via Recovery Email |  Recovery.AdminPasswordReset.RecoveryLink | User gets notified with a link to reset password |
+    | Enable Password Reset via OTP  | Recovery.AdminPasswordReset.OTP  | User gets notified with a one time password to try with SSO login |
```
Comment (gomathyk) [inline en/docs/develop/identity-governance-rest-api.md]: to try out SSO login

File: en/docs/develop/identity-governance-rest-api.md [hunk 15]
```diff
@@ -3,16 +3,164 @@ template: templates/swagger.html
...
+    | Target Host | adaptive_authentication.analytics.receiver | Target Host |
+    | Enable Basic Authentication | adaptive_authentication.analytics.basicAuth.enabled | Enable Basic Authentication |
+    | User ID | adaptive_authentication.analytics.basicAuth.username | Target Host Secured User ID |
+    | Secret | __secret__adaptive_authentication.analytics.basicAuth.password | Target Host Secured Secret |
+    | HTTP Connection Timeout | adaptive_authentication.analytics.HTTPConnectionTimeou | HTTP Connection Timeout in milliseconds |
+    | HTTP Read Timeout | adaptive_authentication.analytics.HTTPReadTimeout | HTTP Read Timeout in milliseconds |
+    | HTTP Connection Request Timeout | adaptive_authentication.analytics.HTTPConnectionRequestTimeout | HTTP Connection Request Timeout in milliseconds |
+    | Hostname verification | adaptive_authentication.analytics.hostnameVerfier | Hostname verification. (STRICT, ALLOW_ALL) |
+
+The APIs mentioned below can be used to retrieve the above mentioned categories, connectors of the categories and properies of the categories and update the property values.
```
Comment (gomathyk) [inline en/docs/develop/identity-governance-rest-api.md]: connectors of the categories, properties of the categories, and update the property values.

File: en/docs/develop/identity-governance-rest-api.md [hunk 16]
```diff
@@ -3,16 +3,164 @@ template: templates/swagger.html
...
+
+??? Note "How to try out the API"
     Do the following to try out the REST APIs with your local instance of WSO2 Identity Server.
```
Comment (gomathyk) [inline en/docs/develop/identity-governance-rest-api.md]: Follow the steps given below to ...

File: en/docs/develop/identity-governance-rest-api.md [hunk 17]
```diff
@@ -3,16 +3,164 @@ template: templates/swagger.html
...
+    | HTTP Read Timeout | adaptive_authentication.analytics.HTTPReadTimeout | HTTP Read Timeout in milliseconds |
+    | HTTP Connection Request Timeout | adaptive_authentication.analytics.HTTPConnectionRequestTimeout | HTTP Connection Request Timeout in milliseconds |
+    | Hostname verification | adaptive_authentication.analytics.hostnameVerfier | Hostname verification. (STRICT, ALLOW_ALL) |
+
+The APIs mentioned below can be used to retrieve the above mentioned categories, connectors of the categories and properies of the categories and update the property values.
+
+??? Note "How to try out the API"
     Do the following to try out the REST APIs with your local instance of WSO2 Identity Server. 
     
     1.  Click on **Authorize** button and provide desired values for authentication.
```
Comment (gomathyk) [inline en/docs/develop/identity-governance-rest-api.md]: Click on **Authorize** and

File: en/docs/develop/identity-governance-rest-api.md [hunk 18]
```diff
@@ -3,16 +3,164 @@ template: templates/swagger.html
...
+    | HTTP Connection Request Timeout | adaptive_authentication.analytics.HTTPConnectionRequestTimeout | HTTP Connection Request Timeout in milliseconds |
+    | Hostname verification | adaptive_authentication.analytics.hostnameVerfier | Hostname verification. (STRICT, ALLOW_ALL) |
+
+The APIs mentioned below can be used to retrieve the above mentioned categories, connectors of the categories and properies of the categories and update the property values.
+
+??? Note "How to try out the API"
     Do the following to try out the REST APIs with your local instance of WSO2 Identity Server. 
     
     1.  Click on **Authorize** button and provide desired values for authentication. 
     2.  Expand the relevant API operation and click the **Try It Out** button.
```
Comment (gomathyk) [inline en/docs/develop/identity-governance-rest-api.md]: click **Try it Out**.

File: en/docs/develop/identity-governance-rest-api.md (hunk 2 above)
Comment (madurangasiriwardena) [reply en/docs/develop/identity-governance-rest-api.md]: Without adding the word "note" it is rendered as a collapsible. So not added the word.

File: en/docs/develop/identity-governance-rest-api.md (hunk 4 above)
Comment (madurangasiriwardena) [reply en/docs/develop/identity-governance-rest-api.md]: Without adding the word "note" it is rendered as a collapsible. So not added the word.

File: en/docs/develop/identity-governance-rest-api.md [hunk 19]
```diff
@@ -3,16 +3,165 @@ template: templates/swagger.html
 ---
 
 # WSO2 Identity Server - governance connectors admin API
+     
+This is the RESTful API for managing identity governance configurations in WSO2 Identity Server. These configurations
+ are grouped in to five main categories. Each category has subcategories which are called connectors, and
```
Comment (gomathyk) [inline en/docs/develop/identity-governance-rest-api.md]: into

File: en/docs/develop/identity-governance-rest-api.md [hunk 20]
```diff
@@ -3,16 +3,165 @@ template: templates/swagger.html
...
+    | Hostname verification | adaptive_authentication.analytics.hostnameVerfier | Hostname verification. (STRICT, ALLOW_ALL) |
+
+The APIs can be used to retrieve the above mentioned categories, connectors of the categories, properties of the categories, and update the property values..
+
+??? Note "How to try out the API"
+    Follow the steps given below to try out the REST APIs with your local instance of WSO2 Identity Server. 
     
-    1.  Click on **Authorize** button and provide desired values for authentication. 
-    2.  Expand the relevant API operation and click the **Try It Out** button.  
+    1.  Click **Authorize** button and provide desired values for authentication.
```
Comment (gomathyk) [inline en/docs/develop/identity-governance-rest-api.md]: please remove the word, "button".

File: en/docs/develop/identity-governance-rest-api.md [hunk 21]
```diff
@@ -3,16 +3,165 @@ template: templates/swagger.html
...
+
+The APIs can be used to retrieve the above mentioned categories, connectors of the categories, properties of the categories, and update the property values..
+
+??? Note "How to try out the API"
+    Follow the steps given below to try out the REST APIs with your local instance of WSO2 Identity Server. 
     
-    1.  Click on **Authorize** button and provide desired values for authentication. 
-    2.  Expand the relevant API operation and click the **Try It Out** button.  
+    1.  Click **Authorize** button and provide desired values for authentication. 
+    2.  Expand the relevant API operation and click the **Try it Out** button.
```
Comment (gomathyk) [inline en/docs/develop/identity-governance-rest-api.md]: Click **Try it Out**. Please remove the word, button.

**Pull request merged**